blockchain/validate.go

```diff
@@ -705,15 +726,16 @@ func (b *BlockChain) checkBlockContext(block *bchutil.Block, prevNode *blockNode
 	}
 
 	uahfActive := block.Height() > b.chainParams.UahfForkHeight
+	ablaActive := block.Height() > b.chainParams.ABLAForkHeight
 
 	// A block must not have more transactions than the max block payload or
 	// else it is certainly over the size limit.
 	// We need to check the blocksize here rather than in checkBlockSanity
 	// because after the Uahf activation it is not longer context free as
 	// the max size depends on whether Uahf has activated or not.
-	maxBlockSize := b.MaxBlockSize(uahfActive)
+	maxBlockSize := b.MaxBlockSize(uahfActive, ablaActive)
 	numTx := len(block.MsgBlock().Transactions)
-	if numTx > maxBlockSize {
+	if uint64(numTx) > maxBlockSize {
```

Comment:
Why do we need this typecast?

Reply:
After the ABLA upgrade, MaxBlockSize() returns uint64 instead of int. Although it is limited to a 32 bit architecture for now. But len does return an int. Can't really compare int and uint64 without casting.